library/src/main/java/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt

```diff
@@ -0,0 +1,155 @@
+package com.getstream.sdk.chat.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations.map
+import androidx.lifecycle.ViewModel
+import com.getstream.sdk.chat.utils.MessageListItemLiveData
+import io.getstream.chat.android.client.logger.ChatLogger.Companion.get
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Command
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.livedata.controller.ChannelController
+
+class MessageInputViewModel(private val cid: String, private val chatDomain: ChatDomain = ChatDomain.instance()) : ViewModel() {
+	val members: LiveData<List<Member>>
+	val commands: LiveData<List<Command>>
+	private var unreadCount: LiveData<Int>
+
+	/**
+	 * The numbers of users currently watching this channel
+	 */
+	private var watcherCount: LiveData<Int>
+
+	/**
+	 * The list of users currently typing
+	 */
+	private var typingUsers: LiveData<List<User>> = MutableLiveData()
+
+
+	private val channelController: ChannelController = chatDomain.useCases.watchChannel.invoke(cid, 30).execute().data()
+	private var online: LiveData<Boolean> = MutableLiveData(false)
+	private var activeThread = MutableLiveData<Message?>(null)
+	private var threadMessages: LiveData<List<Message>>
+	private var threadLoadingMore: LiveData<Boolean>? = null
+
+	private var initialized: LiveData<Boolean> = MutableLiveData(false)
+	private var reachedEndOfPagination: LiveData<Boolean>
+	private var reachedEndOfPaginationThread: LiveData<Boolean>? = null
+	private var loading: LiveData<Boolean> = MutableLiveData(false)
+	private var messageListScrollUp = MutableLiveData(false)
+	private var loadingMore: LiveData<Boolean> = MutableLiveData(false)
+	private var editMessage: MutableLiveData<Message?>
+	private var channelState = MutableLiveData<Channel>()
+
+	private var reads: LiveData<List<ChannelUserRead>> = MutableLiveData()
+	private var entities: MessageListItemLiveData
+	private val logger = get(MessageInputViewModel::class.java.name)
+
+	@get:Deprecated("")
+	val channel: Channel
+		get() = channelController.toChannel()
+
+	fun getEditMessage(): LiveData<Message?> {
+		return editMessage
+	}
+
+	fun setEditMessage(editMessage: Message?) {
```

Comment:
We need to stick to single way of interactive with viewmodels. Either with events and states (as its done by Samuel in `ChannelsViewModel`) or via methods. I prefere first one.